File: job-matcher/scripts/job-artifact-paths.js
#!/usr/bin/env node

require('dotenv').config();

const path = require('path');
const ConfigLoader = require('../src/config/config-loader');

const config = ConfigLoader.load();
const runLabel = buildRunLabel(process.env.JOB_MATCHER_TIMEZONE || 'America/Los_Angeles');
const baseOutputDir = path.dirname(config.output_path);
const baseOutputName = path.basename(config.output_path);
const outputDir = path.join(baseOutputDir, runLabel);
const outputTxt = path.join(outputDir, baseOutputName);
const outputJson = outputTxt.replace(/\.txt$/i, '.json');

function buildRunLabel(timeZone) {
  const parts = new Intl.DateTimeFormat('en-CA', {
    timeZone,
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
    hour: '2-digit',
    minute: '2-digit',
    hour12: false
  }).formatToParts(new Date());

  const map = Object.fromEntries(parts.map(part => [part.type, part.value]));
  return `${map.year}-${map.month}-${map.day}-${map.hour}${map.minute}`;
}

function emit(name, value) {
  process.stdout.write(`${name}=${value}\n`);
}

emit('JOB_MATCHER_RUN_LABEL', runLabel);
emit('JOB_MATCHER_RESUME_PATH', path.resolve(config.resume_path));
emit('JOB_MATCHER_OUTPUT_TXT', path.resolve(outputTxt));
emit('JOB_MATCHER_OUTPUT_JSON', path.resolve(outputJson));
emit('JOB_MATCHER_CACHE_DB', path.resolve(config.ats_cache_db_path));

File: job-matcher/scripts/job_artifact_paths.py
#!/usr/bin/env python3

import os
from datetime import datetime
from pathlib import Path

import yaml


def build_run_label(timezone="America/Los_Angeles"):
    os.environ["TZ"] = timezone
    try:
        # Handle timezone on non-macOS
        if hasattr(time, "tzset"):
            time.tzset()
    except Exception:
        pass

    now = datetime.now()
    return now.strftime("%Y-%m-%d-%H%M")


def main():
    timezone = os.environ.get("JOB_MATCHER_TIMEZONE", "America/Los_Angeles")
    run_label = build_run_label(timezone)

    config_path = Path(__file__).parent.parent / "config" / "jobs.yaml"
    with open(config_path) as f:
        config = yaml.safe_load(f)

    resume_path = Path(config["resume_path"]).resolve()
    base_output_path = Path(config["output_path"]).resolve()
    output_dir = base_output_path.parent / run_label
    output_txt = output_dir / base_output_path.name
    output_json = output_txt.with_suffix(".json")
    cache_db = Path(config.get("ats_cache_db_path", "./output/ats-cache.sqlite")).resolve()

    print(f"JOB_MATCHER_RUN_LABEL={run_label}", flush=True)
    print(f"JOB_MATCHER_RESUME_PATH={resume_path}", flush=True)
    print(f"JOB_MATCHER_OUTPUT_TXT={output_txt}", flush=True)
    print(f"JOB_MATCHER_OUTPUT_JSON={output_json}", flush=True)
    print(f"JOB_MATCHER_CACHE_DB={cache_db}", flush=True)


if __name__ == "__main__":
    main()
